Sources/OpenAPIRuntime/URICoder/Decoding/URIValueFromNodeDecoder+Single.swift

```diff
@@ -17,26 +17,27 @@ import Foundation
 /// A single value container used by `URIValueFromNodeDecoder`.
 struct URISingleValueDecodingContainer {
 
-    /// The coder used to serialize Date values.
-    let dateTranscoder: any DateTranscoder
-
-    /// The coding path of the container.
-    let codingPath: [any CodingKey]
-
-    /// The underlying value.
-    let value: URIParsedValue
+    /// The associated decoder.
+    let decoder: URIValueFromNodeDecoder
 }
 
 extension URISingleValueDecodingContainer {
 
+    /// The underlying value as a single value.
+    var value: URIParsedValue {
+        get throws {
+            try decoder.currentElementAsSingleValue()
+        }
+    }
+
     /// Returns the value found in the underlying node converted to
     /// the provided type.
     /// - Returns: The converted value found.
     /// - Throws: An error if the conversion failed.
     private func _decodeBinaryFloatingPoint<T: BinaryFloatingPoint>(
         _: T.Type = T.self
     ) throws -> T {
-        guard let double = Double(value) else {
+        guard let double = try Double(value) else {
```

Comment:
Because the validation of the underlying value now happens at access time of the throwing `value` property getter, not at container creation time. Delaying it to access time mean we can actually support decoding an arbitrary Decodable type through a single value container, which is required for the rest of it to work. I believe JSON decoder does the same thing.